### DISCUSSION TITLE
No recent ROCM builds

### DISCUSSION THREAD

**Discussion by ross-rosario** [blakeblackshear/frigate#10792] [General]
I use the dev v0.14 releases specifically for ROCM support, and I find that being up-to-date is extremely important. It appears that for > 5 days now there has been no rocm-specific builds, though. Have they been halted?

Also, is there a chat room or a forum where I can drop this kind questions about transient matters?

Thanks.

**Comment by NickM-27** [comment]
See https://github.com/blakeblackshear/frigate/pull/10717

**Reply by ross-rosario** [reply]
That is such a bummer.

**Reply by ross-rosario** [reply]
Well, that's ok. I just want to say how awesome Frigate has been and especially with your continuous and great contributions @NickM-27 🙏

**Comment by avahahn** [comment]
Hey all! I am curious what a path forward for AMD GPU support could look like!
I am assuming it would be prohibitive to relicense Frigate as GPL v3. Otherwise that would be the only change needed to integrate with GPLv3 models.

Are there other models we can use?

**Comment by saket424** [comment]
@NickM-27 
do you have a ready to try config.yml and docker image for the rocm-yolonas branch
https://github.com/blakeblackshear/frigate/tree/rocm-yolonas

docker image pull ghcr.io/blakeblacksear/frigate:rocm-yolonas-7fc54f6
Error response from daemon: Head "https://ghcr.io/v2/blakeblacksear/frigate/manifests/rocm-yolonas-7fc54f6": denied

Thanks in advance

[7fc54f6](https://github.com/blakeblackshear/frigate/commit/7fc54f6aa40df0aa77db1631bf853f1f62ee5a3c)

**Reply by NickM-27** [reply]
No I don't

**Comment by saket424** [comment]
@harakas 
Can you point me to the steps involved in building a rocm docker image
Obviously, I dont know how to proceed and could use your guidance

```
anand@nitro17:~/frigate-stuff/frigate-rocm-yolonas/docker/rocm$ docker build -t asetlur13/frigate:rocm-yolonas .
[+] Building 0.8s (5/5) FINISHED                                 docker:default
 => [internal] load build definition from Dockerfile                       0.0s
 => => transferring dockerfile: 3.43kB                                     0.0s
 => resolve image config for docker-image://docker.io/docker/dockerfile:1  0.2s
 => CACHED docker-image://docker.io/docker/dockerfile:1.4@sha256:9ba7531b  0.0s
 => [internal] load .dockerignore                                          0.0s
 => => transferring context: 2B                                            0.0s
 => ERROR [internal] load metadata for docker.io/library/rocm-prelim-hsa-  0.3s
------
 > [internal] load metadata for docker.io/library/rocm-prelim-hsa-override11.0.0:latest:
------
Dockerfile:101
--------------------
  99 |     
 100 |     #######################################################################
 101 | >>> FROM rocm-prelim-hsa-override$HSA_OVERRIDE as rocm-deps
 102 |     
 103 |     # Request yolov8 download at startup
--------------------
ERROR: failed to solve: failed to resolve source metadata for docker.io/library/rocm-prelim-hsa-override11.0.0:latest: pull access denied, repository does not exist or may require authorization: server message: insufficient_scope: authorization failed
anand@nitro17:~/frigate-stuff/frigate-rocm-yolonas/docker/rocm$
```

**Reply by NickM-27** [reply]
if you want to build it you can just checkout the branch and then run `make rocm-push`

**Reply by saket424** [reply]
there is no such rule rocm-push in the Makefile. There is a push but it errored out. can you enable automated rocm builds for this rocm-yolonas branch? i am specifically interested in the 11.0.0 variant and not the 9.0.0 variant

**Reply by NickM-27** [reply]
No, automated builds will happen on dev, so will need to wait until it is merged into 0.14.1, and then when 0.14.1 is merged into dev.

The make is `push-rocm`

<img width="324" alt="Screenshot 2024-07-21 at 12 05 49 PM" src="https://github.com/user-attachments/assets/8e226362-1f0b-43ba-8bdd-13893fc8ff01">

**Comment by saket424** [comment]
echo 'VERSION = "0.14.0-7fc54f6a"' > frigate/version.py
AMDGPU=gfx1100 HSA_OVERRIDE_GFX_VERSION=11.0.0 HSA_OVERRIDE=1 docker buildx bake --file=docker/rocm/rocm.hcl --set rocm.tags=xyz/frigate:rocm-yolonas-7fc54f6a-rocm-gfx1100 rocm

**Comment by saket424** [comment]
Even though i used harkas model_cache and changed the model_type i get this error. a working yolonas config would be greatly appreciated

```
model:
  labelmap_path: /config/model_cache/yolov8/labels.txt
  model_type: yolox
```

```
frigate  | 2024-07-21 20:27:43.984582339  *************************************************************
frigate  | 2024-07-21 20:27:43.984583751  ***    Your config file is not valid!                     ***
frigate  | 2024-07-21 20:27:43.984584864  ***    Please check the docs at                           ***
frigate  | 2024-07-21 20:27:43.984587699  ***    https://docs.frigate.video/configuration/index     ***
frigate  | 2024-07-21 20:27:43.984610111  *************************************************************
frigate  | 2024-07-21 20:27:43.984611293  *************************************************************
frigate  | 2024-07-21 20:27:43.984612916  ***    Config Validation Errors                           ***
frigate  | 2024-07-21 20:27:43.984613968  *************************************************************
frigate  | 2024-07-21 20:27:43.984630460  model.model_type: Input should be 'ssd', 'yolox' or 'yolonas'
frigate  | 2024-07-21 20:27:43.984631401  *************************************************************
frigate  | 2024-07-21 20:27:43.984632353  ***    End Config Validation Errors                       ***
frigate  | 2024-07-21 20:27:43.984633275  *************************************************************
```

**Reply by NickM-27** [reply]
If you are using my branch then only a yolonas model can be used.

**Reply by saket424** [reply]
oh i just changed yolox to yolonas and it worked. gracias

**Comment by saket424** [comment]
@harakas 
i don't want to jinx it just yet -- this docker image I built with NickM's branch  is running better than the 5 month old docker image. no restarts on the AMD 8845 laptop

**Comment by saket424** [comment]
@harakas 
I am thoroughly confused. The same setup that was working yesterday is no longer working. I get the following error after the model load is completed. Also I noticed that even though I had commented out privileged, the docker is containing folders owned by root. is this normal?

```

frigate  | 2024-07-22 11:16:55.331990656  [2024-07-22 11:16:54] frigate.detectors.plugins.rocm INFO    : AMD/ROCm: loading parsed model from /config/model_cache/yolov8/yolov8s_320x320.mxr
frigate  | 2024-07-22 11:16:55.332165004  [2024-07-22 11:16:55] frigate.detectors.plugins.rocm INFO    : AMD/ROCm: model loaded
frigate  | 2024-07-22 11:16:58.176893641  Process detector:rocm:
frigate  | 2024-07-22 11:16:58.177713254  Traceback (most recent call last):
frigate  | 2024-07-22 11:16:58.177724505    File "/usr/lib/python3.9/multiprocessing/process.py", line 315, in _bootstrap
frigate  | 2024-07-22 11:16:58.177725618      self.run()
frigate  | 2024-07-22 11:16:58.177734264    File "/usr/lib/python3.9/multiprocessing/process.py", line 108, in run
frigate  | 2024-07-22 11:16:58.177735145      self._target(*self._args, **self._kwargs)
frigate  | 2024-07-22 11:16:58.177735987    File "/opt/frigate/frigate/object_detection.py", line 125, in run_detector
frigate  | 2024-07-22 11:16:58.177736899      detections = object_detector.detect_raw(input_frame)
frigate  | 2024-07-22 11:16:58.177737971    File "/opt/frigate/frigate/object_detection.py", line 75, in detect_raw
frigate  | 2024-07-22 11:16:58.177765362      return self.detect_api.detect_raw(tensor_input=tensor_input)
frigate  | 2024-07-22 11:16:58.177766865    File "/opt/frigate/frigate/detectors/plugins/rocm.py", line 140, in detect_raw
frigate  | 2024-07-22 11:16:58.177768538      detector_result = self.model.run({model_input_name: tensor_input})[0]
frigate  | 2024-07-22 11:16:58.177778337  RuntimeError: /long_pathname_so_that_rpms_can_package_the_debug_info/src/extlibs/AMDMIGraphX/src/program.cpp:461: operator(): Incorrect shape {uint8_type, {1, 3, 320, 320}, {307200, 102400, 320, 1}} for parameter: images should be: float_type, {1, 3, 320, 320}, {307200, 102400, 320, 1}
frigate  | 2024-07-22 11:17:01.623025143  [INFO] Starting go2rtc healthcheck service...
frigate  | 2024-07-22 11:17:14.731857283  [2024-07-22 11:17:14] frigate.watchdog               INFO    : Detection appears to be stuck. Restarting detection process...

```

**Comment by saket424** [comment]
@NickM-27 
the containerseems to be expecting float onnx values instead of int8 . do you have any suggestions ?

**Comment by ross-rosario** [comment]
Let's wait for the 14.1 release where @NickM-27 is planning to include this. Thanks for all the testing and for including the minority of users that dare to use an AMD card 🙏 .
